### PR TITLE
feat: autolabel fields when omitted

### LINKED ISSUE #37: Auto-generate field labels if none specified
# Feature Request

Auto-generate field labels based on the field's `name`.  Allow for labels to be disabled by passing `false` to the `label` field property.

## Use-Case

It gets a bit tedious to specify field labels over and over, when the `name` of the field could probably be used and formatted properly to generate a field label automatically.

By generating field labels automatically during field sanitization, there will be less configuration necessary by the developer.

## Additional Detail

We need to make sure to adjust the field schema validation _**and**_ field type(s) to support either strings or booleans to be passed to the `label` property of all fields that support labels.

We also need to make sure to update Field documentation with the corresponding API changes.

### PR DESCRIPTION
**Pull request by denolfe**
- Field labels get labeled automatically if left undefined.
- `label: false` opts out of getting labeled
- `label: 'My Custom Label'` overrides auto-labeling

Resolves #37

### CODE REVIEW

**Comment by denolfe** [comment]
Groups must have a label for pill text.

**Comment by denolfe** [comment]
Ready for review

File: src/admin/components/forms/DraggableSection/ActionPanel/types.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { Block } from '../../../../../fields/config/types';
 
 export type Props = {
-  label: string
+  label: string | false
```
Comment (jmikrut) [inline src/admin/components/forms/DraggableSection/ActionPanel/types.ts:4]: I believe this should always have a label as well. Within the `ActionPanel`, we use labels for "Add Row" and "Remove Row" / etc. It should be sanitized at this point I believe.

File: src/graphql/schema/buildMutationInputType.ts [hunk 2]
```diff
@@ -68,7 +68,7 @@ function buildMutationInputType(name: string, fields: Field[], parentName: strin
       let type: PayloadGraphQLRelationshipType = GraphQLString;
 
       if (Array.isArray(relationTo)) {
-        const fullName = `${combineParentName(parentName, field.label)}RelationshipInput`;
+        const fullName = `${combineParentName(parentName, !field.label ? '' : field.label)}RelationshipInput`;
```
Comment (jmikrut) [inline src/graphql/schema/buildMutationInputType.ts:71]: At this point, this is where GraphQL types are being named. We should be using the sanitized label for the field here. Using an empty string will create the potential for naming conflicts.

File: src/graphql/schema/buildMutationInputType.ts [hunk 3]
```diff
@@ -91,14 +91,14 @@ function buildMutationInputType(name: string, fields: Field[], parentName: strin
       return { type: field.hasMany ? new GraphQLList(type) : type };
     },
     array: (field: ArrayField) => {
-      const fullName = combineParentName(parentName, field.label);
+      const fullName = combineParentName(parentName, !field.label ? '' : field.label);
```
Comment (jmikrut) [inline src/graphql/schema/buildMutationInputType.ts:94]: Same here

File: src/graphql/schema/buildMutationInputType.ts [hunk 4]
```diff
@@ -91,14 +91,14 @@ function buildMutationInputType(name: string, fields: Field[], parentName: strin
...
       let type: GraphQLType | GraphQLList<GraphQLType> = buildMutationInputType(fullName, field.fields, fullName);
       type = new GraphQLList(withNullableType(field, type, forceNullable));
       return { type };
     },
     group: (field: GroupField) => {
       const requiresAtLeastOneField = field.fields.some((subField) => (subField.required && !subField.localized));
-      const fullName = combineParentName(parentName, field.label);
+      const fullName = combineParentName(parentName, !field.label ? '' : field.label);
```
Comment (jmikrut) [inline src/graphql/schema/buildMutationInputType.ts:101]: Same here

File: src/admin/components/forms/field-types/Array/types.ts [hunk 5]
```diff
@@ -7,6 +7,7 @@ export type Props = Omit<ArrayField, 'type'> & {
   path?: string
   fieldTypes: FieldTypes
   permissions: FieldPermissions
+  label: string
```
Comment (jmikrut) [inline src/admin/components/forms/field-types/Array/types.ts:10]: This should also be allowed to be set to false as well, right? As the Blocks type?

File: src/graphql/schema/buildMutationInputType.ts [hunk 6]
```diff
@@ -68,7 +68,7 @@ function buildMutationInputType(name: string, fields: Field[], parentName: strin
       let type: PayloadGraphQLRelationshipType = GraphQLString;
 
       if (Array.isArray(relationTo)) {
-        const fullName = `${combineParentName(parentName, field.label)}RelationshipInput`;
+        const fullName = `${combineParentName(parentName, field.label as string)}RelationshipInput`;
```
Comment (jmikrut) [inline src/graphql/schema/buildMutationInputType.ts:71]: What happens if this field's label is set to `false`? Would this not create a proper field name, or worse, would it stringify `false` as the name? We should fall back to `field.name` here instead I think.

**Pull request merged**